packages/camera/camera_web/lib/src/camera_settings.dart

```diff
@@ -107,6 +109,65 @@ class CameraSettings {
     }
   }
 
+  /// Returns the zoom level capability for the given [camera].
+  ///
+  /// Throws a [CameraWebException] if the zoom level is not supported
+  /// or the camera has not been initialized or started.
+  ZoomLevelCapability getZoomLevelCapabilityForCamera(
+    Camera camera,
+  ) {
+    final mediaDevices = window?.navigator.mediaDevices;
+    final supportedConstraints = mediaDevices?.getSupportedConstraints();
+    final zoomLevelSupported =
+        supportedConstraints?[ZoomLevelCapability.constraintName] ?? false;
+
+    if (!zoomLevelSupported) {
+      throw CameraWebException(
+        camera.textureId,
+        CameraErrorCode.zoomLevelNotSupported,
+        'The zoom level is not supported in the current browser.',
+      );
+    }
+
+    final videoTracks = camera.stream?.getVideoTracks() ?? [];
+
+    if (videoTracks.isNotEmpty) {
+      final defaultVideoTrack = videoTracks.first;
+
+      /// The zoom level capability is represented by MediaSettingsRange.
+      /// See: https://developer.mozilla.org/en-US/docs/Web/API/MediaSettingsRange
+      final zoomLevelCapability = defaultVideoTrack
+              .getCapabilities()[ZoomLevelCapability.constraintName] ??
+          {};
+
+      // The zoom level capability is a nested JS object, therefore
+      // we need to access its properties with the js_util library.
+      // See: https://api.dart.dev/stable/2.13.4/dart-js_util/getProperty.html
+      final minimumZoomLevel = js_util.getProperty(zoomLevelCapability, 'min');
+      final maximumZoomLevel = js_util.getProperty(zoomLevelCapability, 'max');
```

Comment:
Does this work if you make zoomLevelCapability a `dynamic` and you just access `.min` on it?

Reply:
Unfortunately not, it throws:

```NoSuchMethod Error: method not found: '$index* (J.getInterceptor$asx(...). $index is not a function)```

when trying to access a nested JS property either with `zoomLevelCapability.min` or `zoomLevelCapability['min']`.

Reply:
:'(